Import default template args

martong opened the pull request:
Fixes #310

Commit:
Import default template args

Xazax-hun commented on lib/AST/ASTImporter.cpp:
```diff
@@ -4174,7 +4174,7 @@ Decl *ASTNodeImporter::VisitTemplateTypeParmDecl(TemplateTypeParmDecl *D) {
   
   // FIXME: Import default argument.
-  return TemplateTypeParmDecl::Create(Importer.getToContext(),
+  auto ToD = TemplateTypeParmDecl::Create(Importer.getToContext(),
```
Fixme above is no longer relevant.

Xazax-hun reviewed (commented):
What about value templates with default arguments?

martong commented:
Added new test for this case:
```
      template<typename T0, typename T1 = int> class A;
      template<typename T0 = int, typename T1> class A;
```

Added new test for VarTemplateDecl.

balazske reviewed (commented):
It looks good, but only type kind of template arguments are used at the test. Does it work similarly with value and template type of arguments?

gamesh411 commented:
Can one of the admins verify this patch?

martong commented:
Not important for a while, because all specializations/instantiations have the concrete parameter. We did not see any error caused by this.
